collated/main/CindyTsai1.md

```diff
@@ -0,0 +1,536 @@
+     * @param tags
+     */
+    public ArrayList<String> findTags(String tags) {
+        ObservableList<ReadOnlyPerson> personList = model.getAddressBook().getPersonList();
+        String[] tagKeyword = tags.split(" ");
+        ArrayList<String> tagList = new ArrayList<>();
+        for (ReadOnlyPerson person : personList) {
+            for (String keyword : tagKeyword) {
+                for (Tag tagging : person.getTags()) {
+                    String tag1 = tagging.toString().substring(1, tagging.toString().length()).toLowerCase();
+                    keyword = keyword.toLowerCase();
+                    if (tag1.contains(keyword)) {
+                        tagList.add(person.getName().toString(0));
+                    }
+                }
+            }
+        }
+        return tagList;
+    }
+
+    /**
+     * Search for persons that contain the {@String keyword} in their birthday
+     * @param birthday
+     */
+    public ArrayList<String> findBirthday(String birthday) {
+        ObservableList<ReadOnlyPerson> personList = model.getAddressBook().getPersonList();
+        String[] birthdayKeyword = birthday.split(" ");
+        ArrayList<String> birthdayList = new ArrayList<>();
+        for (ReadOnlyPerson person : personList) {
+            for (String keyword : birthdayKeyword) {
+                String birthdays = person.getBirthday().toString().substring(2, 4);
+                if (birthdays.equals(keyword)) {
+                    birthdayList.add(person.getName().toString(0));
+                }
+            }
+        }
+        return birthdayList;
+    }
+}
+```
+###### /java/seedu/address/logic/commands/SuggestCommand.java
+``` java
+/**
+ * Suggests a correct command
+ */
+public class SuggestCommand extends Command {
+    public static final String MESSAGE_SUCCESS = "Do you mean %1$s?";
+
+    private final String possibleCommand;
+
+    /**
+     * Creates an SuggestCommand to suggest the specified {@code String} for command
+     */
+    public SuggestCommand(String command) {
+        this.possibleCommand = command;
+    }
+
+    @Override
+    public CommandResult execute() throws CommandException {
+        throw new CommandException(String.format(MESSAGE_SUCCESS, possibleCommand));
+    }
+}
+```
+###### /java/seedu/address/logic/commands/UniqueCommandList.java
+``` java
+/**
+ * A list of all command words, including alias and secondary word
+ */
+public class UniqueCommandList {
+    private static ArrayList<String> commandList;
+    private static TreeSet<String> possibleCommandList;
+
+    public static ArrayList<String> getCommandList() {
+        commandList = new ArrayList<>();
+        commandList.add(AddCommand.COMMAND_WORD);
+        commandList.add(AddCommand.COMMAND_ALIAS);
+        commandList.add(AddCommand.COMMAND_SECONDARY);
+        commandList.add(ClearCommand.COMMAND_WORD);
+        commandList.add(ClearCommand.COMMAND_ALIAS);
```

Comment:
I wonder if this would be a more concise alternative ...
```java
commandList = new ArrayList<>(Arrays.asList(  
        AddCommand.COMMAND_WORD, AddCommand.COMMAND_ALIAS, AddCommand.COMMAND_SECONDARY,
        ClearCommand.COMMAND_WORD, ClearCommand.COMMAND_ALIAS, ... ));
```